pkg/reconciler/artifactbuildrequest/artifactbuildrequest.go

```diff
@@ -59,43 +67,91 @@ func (r *ReconcileArtifactBuildRequest) Reconcile(ctx context.Context, request r
-		return r.handleStateDiscovered(ctx, abr)
+		return r.handleStateNew(ctx, &abr)
+	case v1alpha1.ArtifactBuildRequestStateDiscovering:
+		return r.handleStateDiscovering(ctx, &abr)
 	case v1alpha1.ArtifactBuildRequestStateComplete:
-		return r.handleStateComplete(ctx, abr)
+		return r.handleStateComplete(ctx, &abr)
+	case v1alpha1.ArtifactBuildRequestStateBuilding:
+		return r.handleStateBuilding(ctx, &abr)
 	}
 	return reconcile.Result{}, nil
 }
 
-func (r *ReconcileArtifactBuildRequest) handleStateNew(ctx context.Context, abr v1alpha1.ArtifactBuildRequest) (reconcile.Result, error) {
+func (r *ReconcileArtifactBuildRequest) handleStateNew(ctx context.Context, abr *v1alpha1.ArtifactBuildRequest) (reconcile.Result, error) {
+
 	// create task run
 	tr := pipelinev1beta1.TaskRun{}
 	tr.Spec.TaskRef = &pipelinev1beta1.TaskRef{Name: "lookup-artifact-location", Kind: pipelinev1beta1.NamespacedTaskKind}
 	tr.Namespace = abr.Namespace
 	tr.GenerateName = abr.Name + "-scm-discovery-"
-	tr.Labels = map[string]string{TaskRunLabel: ""}
+	tr.Labels = map[string]string{ArtifactBuildRequestIdLabel: string(abr.UID), TaskRunLabel: ""}
 	tr.Spec.Params = append(tr.Spec.Params, pipelinev1beta1.Param{Name: "GAV", Value: pipelinev1beta1.ArrayOrString{Type: pipelinev1beta1.ParamTypeString, StringVal: abr.Spec.GAV}})
-	if err := controllerutil.SetOwnerReference(&abr, &tr, r.scheme); err != nil {
+	if err := controllerutil.SetOwnerReference(abr, &tr, r.scheme); err != nil {
 		return reconcile.Result{}, err
 	}
 	if err := r.client.Create(ctx, &tr); err != nil {
 		return reconcile.Result{}, err
 	}
 	abr.Status.State = v1alpha1.ArtifactBuildRequestStateDiscovering
-	if err := r.client.Status().Update(ctx, &abr); err != nil {
+	if err := r.client.Status().Update(ctx, abr); err != nil {
 		return reconcile.Result{}, err
 	}
 	return reconcile.Result{}, nil
 }
 
-func (r *ReconcileArtifactBuildRequest) handleStateDiscovered(ctx context.Context, abr v1alpha1.ArtifactBuildRequest) (reconcile.Result, error) {
+func (r *ReconcileArtifactBuildRequest) handleStateDiscovering(ctx context.Context, abr *v1alpha1.ArtifactBuildRequest) (reconcile.Result, error) {
+	log := log2.FromContext(ctx)
+	log.Info("Discovering: ", "abr", abr.Name)
+	//lets look up our discovery task
+	listOpts := &client.ListOptions{
+		Namespace:     abr.Namespace,
+		LabelSelector: labels.SelectorFromSet(map[string]string{ArtifactBuildRequestIdLabel: string(abr.UID)}),
+	}
+	trl := pipelinev1beta1.TaskRunList{}
+	err := r.client.List(ctx, &trl, listOpts)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	var tr *pipelinev1beta1.TaskRun
+	//there should only be one, be guard against multiple
+	for _, current := range trl.Items {
+		if tr == nil || tr.CreationTimestamp.Before(&current.CreationTimestamp) {
+			tr = &current
+		}
+	}
+	if tr == nil {
+		//no pipeline, very odd
+		//move back to new
+		abr.Status.State = v1alpha1.ArtifactBuildRequestStateNew
```

Comment:
can you put a TODO comment to remind me to generate an event when this happens

I'll handle dealing with events in a separate PR